### PR TITLE
test(http-client): add e2e for http-client invoking token threshold

### PR DESCRIPTION
**Pull request by daveadams56**
# JIRA Ticket

[SDKS-1718: Extend coverage of e2e to include Http Client](https://bugster.forgerock.org/jira/browse/SDKS-1718)

# Description

Description
The recent change to implement access token expiry threshold in the JS SDK ([SDKS-1641](https://bugster.forgerock.org/jira/browse/SDKS-1641)), contained additional e2e testing to exercise the change directly with the TokenManager. This PR focuses on extending the e2e testing further to also exercise the token threshold functionality by invoking the Http Client. Testing both flows makes our testing more representative of how the token threshold code will be used in real implementations.

# Type of Change

Please Delete options that are not relevant

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

This is an extension to the e2e test apps and test suite.

# Definition of Done

Check all that apply

- [x] Acceptance criteria is met.
- [x] All tasks listed in the user story have been completed.
- [x] Coded to standards.
- [ ] Code peer-reviewed.
- [ ] Ensure backward compatibility (special attention).
- [ ] API reference docs is updated.
- [ ] Unit tests are written.
- [ ] Integration tests are written.
- [x] e2e tests are written.
- [ ] CI build passing on the feature branch.
- [ ] Functional spec is written/updated
- [ ] contains example code snippets.
- [ ] Change log updated.
- [ ] Documentation story is created and tracked.
- [ ] UI is completed or ticket is created.
- [ ] Demo to PO and team.
- [ ] Tech debts and remaining tasks are tracked in separated ticket(s).

## Documentation

- [ ] Acceptance criteria met
- [ ] Spell-check run
- [ ] Peer reviewed
- [ ] Proofread

### CODE REVIEW

**Review by ryanbas21 — APPROVED** [review]

File: e2e/autoscript-apps/src/authn-oauth/autoscript.ts [hunk 1]
```diff
@@ -130,6 +133,28 @@ function autoscript() {
         },
       ),
       rxDelay(delay),
+      mergeMap(
+        (tokens) => {
+          console.log('Retrieve a resource');
+          return forgerock.HttpClient.request({
```
Comment (cerebrl) [inline e2e/autoscript-apps/src/authn-oauth/autoscript.ts:139]: Can we pull this out of the `authn-oauth` autoscript and test suite and create a new pair of autoscripts and suites for testing `HttpClient` specifically, or, better yet, the user-flow that it represents? My concern is that we are technically testing the `HttpClient` module. Now, we may be also testing modules that `HttpClient` pulls in internally. But, if we considered all the individual modules that something we are testing pulls in, our naming would be incredibly hard to parse.

For most testing, we should name it according to what we are directly importing and calling in our tests. For E2E, I like to refer to the test by the user-flow that it's intending to automate. The `HttpClient` isn't really directly related to authentication or OAuth, even if it uses shared modules with that functionality. So, it being included in `authn-oauth` is not semantically correct.

This is why we have autoscripts and suites for `authz-txn-basic` and such as that represents the flow we are testing. Can we name these more along those lines? It could be something as simple as `protected-request` or `authz-token` ... something describing the flow. I wouldn't be too specific though, like including "expiry" or "expiry-time" in the name as that would be too limiting.

Hit me up if you want to chat about it. Naming is ridiculously hard.

**Review by cerebrl — CHANGES_REQUESTED** [review]

**Review by cerebrl — APPROVED** [review]
Ran locally and looks good.

**Review by ryanbas21 — APPROVED** [review]
Reapproving
